### PR TITLE
[wip] Initial implementation

### PR DESCRIPTION
**Pull request by kesla**
Break out embeds implementation from html-to-article-json. In the future this repo will include render-methods as well - so you can both parse & render embeds!

R = @iefserge
T = major

TODO:
- [x] Travis
- [ ] Readme
- [ ] Update https://www.npmjs.com/package/html-to-article-json to use this repo

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -0,0 +1,381 @@
+import test from 'ava';
+import 'babel-core/register';
```
Comment (iefserge) [inline test.js:2]: shouldn't be needed when using ava

File: lib/types/custom.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  }
+
+  const src = elm.getAttribute('src');
+  const {width, height} = getDimensions(elm);
+  if (!src || !width || !height) {
+    return null;
+  }
+
+  const parsed = url.parse(src, false, true);
+  const secure = parsed.protocol === 'https:';
```
Comment (iefserge) [inline lib/types/custom.js]: this should handle protocol-relative urls too

**Comment by iefserge** [comment]
Looks great!

File: lib/types/custom.js (hunk 2 above)
Comment (kesla) [inline lib/types/custom.js]: oh right, thanks. Saw now that I was exporting this from an old commit in html-to-article-json :)

File: test.js (hunk 1 above)
Comment (kesla) [inline test.js:2]: It is needed. https://www.npmjs.com/package/ava#transpiling-imported-modules

Comment (kesla) [inline test.js:2]: I _guess_ I could configure it in the package.json instead, if you think that's nicer?

Comment (iefserge) [inline test.js:2]: ah right, you're importing other es6 modules

Comment (iefserge) [inline test.js:2]: was confused about this in ava, weird that it doesn't support importing es6 out of the box

**Pull request merged**